#! /usr/bin/env python

PACKAGE='header_manipulation'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_id_new", str_t, 0, "Frame id replacing the old one.", "/laser1_frame_cal")
gen.add("frame_id_to_replace", str_t, 0, "Frame id to be replaced if found in a header.", "/laser1_frame")
gen.add("msg_delay_milliseconds", double_t, 0, "Timespan the message will be delayed.", 0.0, 0.0, 600.0)
gen.add("time_offset_milliseconds", double_t, 0, "Time offset added to the header timestamp of the msg.", 2000.0, -1000.0, 1000.0)
gen.add("publish_retry_rate", double_t, 0, "Max rate at which the publish msgs loop will try to publish in all concurrent publishMsg Threads.", 100.0, 0.01, 1000.0)

exit(gen.generate(PACKAGE, "HeaderManipulation", "HeaderManipulation"))
